niceml/data/datasets/dfdataset.py

```diff
@@ -100,19 +220,34 @@ def get_all_data(self):
     def extract_data(self, cur_indexes, cur_input):
         """extracts data"""
         cur_key = cur_input["key"]
-        cur_data = self.dataframe.loc[cur_indexes, cur_key]
-        if cur_input["type"] == "categorical":
+        cur_data = self.data.iloc[cur_indexes][cur_key]
+        if cur_input["type"] == FeatureType.CATEGORICAL:
             cur_data = to_categorical(cur_data, cur_input["value_count"])
         return cur_data
 
     def __getitem__(self, index):
+        """
```

Comment:
Please rewrite this docstring as it is very confusing without looking at the code and does not represent the functionality very well.

Reply:
I've rewritten the docstring

Reply:
I still find this to be more complecated than it has to be.
Suggestion:
The __getitem__ function returns the data batch in the size of 'self.batch_size', starting with the indexed data entry (=index). It is called when the `DfDataset` is accessed, using the notation `self[index]` (while training a model).

Reply:
Done